uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -39,12 +41,15 @@ define([
       region: null,
       contacts_with_leaves: false
     };
+    vm.months = ['January', 'February', 'March', 'April', 'May', 'June',
+      'July', 'August', 'September', 'October', 'November', 'December'];
     vm.loading = {
       calendar: false,
+      months: vm.months.map(function () {
```

Comment:
The de-coupling of the loading flag and the actual thing that the flag is referring to is not optimal, it makes it harder to follow and the structure is not as elegant and simple as it could be. Ideally you would have something like this
```js
[{
  label: 'January',
  leaveRequests: [ /* .. */ ],
  loading: true
},
{
  label: 'February',
  leaveRequests: [ /* .. */ ],
  loading: true
}]
```
where on the object itself you are storing the `loading` state, without relying on some other variable loosely related to it

The `vm.loading` property is used to mark the loading state of sections of the page, rather than individual objects like a month is